### ISSUE TITLE
Multi-class classification 

### ISSUE THREAD

**Issue by ruchaa01** [quangnhat185/Machine_learning_projects#6]
Reference to your 'How to train YOLOv3 on Google COLAB to detect custom objects (e.g: Gun detection).'
In case I want to train the network for detection of multiple classes, do I need to make any other changes in yolov3_custom_train.cfg apart from changing the number of filters (line 603, 689, 776 ) and classes (line 610, 696, 783. Do lines # Line 8 & 9: width = 416, height = 416 , # Line 20: max_batches = 6000, # Line 22: steps = 5400 remain the same?
Foe eg: I want to detect 5 classes. My yolo.names file will contain names of those classes on separate lines. My yolo.data file will contain classes=5. In yolov3_custom_train.cfg file will have following changes: classes = 5 (line 610, 696,783) and filters = 30 (line 603, 689,776). Apart from these do I need to make other changes?

**Comment by quangnhat185** [comment]
Hello @ruchaa01,

- `max_batches` indicates the maxium iterations of training cycle. It really depends on how is your training result. If the loss value stays constant after several cycles then it is meaningless to train more. 
- `steps` indicates the number of interations in which learning rate would be reduced. 

Normally with a small number of classes, you can remain the same `max_batches`and `steps` . In case you found there is no significant reduction in loss value, you can decreased  `steps` (reduce learning rate sooner). In case you want to train longer (Colab allows only 12 hour training continuously, so I don't think you can train more than 10,000 cycles), you can increase the `max_batches`. 

You can refer for more details  about these parameters from yolov3's author description [page](https://github.com/AlexeyAB/darknet/wiki/CFG-Parameters-in-the-%5Bnet%5D-section).
